### PR TITLE
check return values with new unit tests

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**

### CODE REVIEW

File: tests/api.c [hunk 1]
```diff
@@ -5940,14 +6002,15 @@ static int test_wc_AesCbcEncryptDecrypt (void)
     }
         printf(resultFmt, cbcDWK == 0 ? passed : failed);
 #endif
-    return 0;
+    return ret;
```
Comment (cconlon) [inline tests/api.c]: Is returning "ret" correct here?  Looks like this section of code also uses "cbcE", "cbcD", and "cbcDWK" to hold error state.

File: tests/api.c [hunk 2]
```diff
@@ -6232,7 +6294,7 @@ static int test_wc_AesGcmEncryptDecrypt (void)
     printf(resultFmt, gcmD == 0 ? passed : failed);
 
 #endif
-    return 0;
+    return ret;
```
Comment (cconlon) [inline tests/api.c]: Do we need to factor in if "gcmE" or "gcmD" is in a failure state here too?

File: tests/api.c [hunk 3]
```diff
@@ -6652,7 +6714,7 @@ static int test_wc_CamelliaEncryptDecryptDirect (void)
     printf(resultFmt, camD == 0 ? passed : failed);
 
 #endif
-    return 0;
+    return ret;
```
Comment (cconlon) [inline tests/api.c]: Do we need to factor in "camD" and "camE" here for SSL_FATAL_ERROR?

File: tests/api.c [hunk 4]
```diff
@@ -6743,7 +6805,7 @@ static int test_wc_CamelliaCbcEncryptDecrypt (void)
     printf(resultFmt, camCbcD == 0 ? passed : failed);
 
 #endif
-    return 0;
+    return ret;
```
Comment (cconlon) [inline tests/api.c]: Do we need to factor in "camCbcE" and "camCbcD" for SSL_FATAL_ERROR here?

**Comment by JacobBarthelmeh** [comment]
Thanks for the comments Chris, am adding a commit to check for the intermediate values.

**Pull request merged**